### ISSUE TITLE
Есть проблема с импортом корпоративного действия от MAC 3.06.2020

### ISSUE THREAD

**Issue by President** [KonishchevDmitry/investments#27]
Такое корпоративное действие не парсится
E: Error while reading ...  broker statement: Failed to parse ("Corporate Actions", "Data", "Stocks", "USD", "2020-06-03", "2020-06-03, 20:20:00", "MAC (US5543821012) Cash Dividend USD 0.10, STOCK DIVIDEND US5543821012 548275673 FOR 10000000000 (MAC, MACERICH CO/THE, US5543821012)", "5", "0", "42.95", "0", "") record: Unsupported corporate action: "MAC (US5543821012) Cash Dividend USD 0.10, STOCK DIVIDEND US5543821012 548275673 FOR 10000000000 (MAC, MACERICH CO/THE, US5543821012)".

3.06.2020 MAC выплачивало дивиденды в размере 0.5$ на акцию в форме 0.1$ на акцию плюс 0.4$ в форме акций MAC с округлением до целых акций и дозачислением остатка тоже в форме $$
Вот здесь подробности https://money.tmx.com/en/quote/MAC:US/news/8482400016399050/Macerich_Announces_Results_Of_Dividend_Payable_On_June_3_2020

В отчете броке в итоге есть две строки:
- строка с дивидендами в $ на которую начислен полный налог 10% от всех дивидендов
- строка с корпоративным действием о получении акций причем цена в день получения уже немного изменилась

Хотя это наверное не частое явление но хорошо бы поддержать и такие корп действия. Иначе невозможно пользоваться программой

**Comment by KonishchevDmitry** [comment]
А у вас есть понимание, как правильно декларировать такие вещи с точки зрения нашего законодательства: дивиденд + покупка или покупка акций по нулевой цене?

**Comment by President** [comment]
Точно не уверен. Задекларировал как получение дохода в размере стоимости акций в день их получения. В отчете брокера эта же сумма была указана для этих полученных акций. И кажется это логично. Ведь мы доход учитываем на дату его получения а не начисления. Поэтому мне кажется что с акциями также должно быть. Но налоговая пока мой отчет не проверила и возможны сюрпризы.

**Comment by titov-vv** [comment]
А вот непонятно как это трактовать.
С одной стороны - как @President написал выше можно как получение дохода в натуральной форме. Т.е. да, принимаем, что акции мы получили по рыночной цене и с этой суммы должны уплатить налог (а потом эту сумму учесть как затраты на покупку бумаг).
Но с другой - акции же не откуда-то взялись, а были выпущены новые, т.е. получается они размыли стоимость старых и  "старые" акции стали стоить дешевле (т.е. сумма, изначально потраченная на покупку акций теперь делится на большее количество акций...)
Я у себя в JAL по второму подходу сделал. Но тоже допускаю, что налоговая может трактовать это по первому варианту.
Что касается того, что налог был удержан в US - то у них немного под другому. Насколько я смог понять из их tax code - этот случай, это Choice Dividend - т.к. на самом деле там было голосование и можно было выбрать что тебе будет зачислено - деньги или бумаги. Именно в этом случае у них бумаги тоже облагаются налогом, как дивиденды. Если бы был просто выплачен дивиденд акциями, то они не взимают налог и размывают cost basis. В нашем НК такого различия нет, собственно поэтому я склонился ко второму варианту с размытием стоимости в своей реализации.

**Comment by KonishchevDmitry** [comment]
@President, приложите, пожалуйста, еще соответствующие строчки из `Dividends` и `Withholding Tax`, чтобы было понятно как накладывать одно на другое.

В идеале - прислать мне сами отчеты на почту, чтобы можно было поддержку не вслепую делать.

**Comment by President** [comment]
Вот все строчки которые я нашел которые относятся к этому событию:
Dividends,Data,USD,2020-06-03,"MAC (US5543821012) Cash Dividend USD 0.50, STOCK DIVIDEND US5543821012 548275673 FOR 10000000000 (Ordinary Dividend)",56.47

Withholding Tax,Data,USD,2020-06-03,"MAC (US5543821012) Cash Dividend USD 0.10, STOCK DIVIDEND US5543821012 548275673 FOR 10000000000 - US Tax",-5,

Change in Dividend Accruals,Data,Stocks,USD,MAC,2020-06-03,2020-04-21,2020-06-03,100,-5,0,0.5,-50,-45,Re

Trades,Data,Order,Stocks,USD,MAC,"2020-06-03, 09:50:04",5,8.59,8.59,-42.95,0,42.95,0,0,O

Прислать весь отчет, к сожалению, не могу. Но нужные строчки готов предоставить

**Comment by KonishchevDmitry** [comment]
Так, с reverse split'ами в соседнем тикете вроде разобрался - пожалуй, теперь смогу уделить внимание и этому тикету. :)

Я ведь правильно понимаю, что по сути вам было выплачено $56.47:
```
Dividends,Data,USD,2020-06-03,"MAC (US5543821012) Cash Dividend USD 0.50, STOCK DIVIDEND US5543821012 548275673 FOR 10000000000 (Ordinary Dividend)",56.47
```
со всей суммы был удержан налог 10%:
```
Withholding Tax,Data,USD,2020-06-03,"MAC (US5543821012) Cash Dividend USD 0.10, STOCK DIVIDEND US5543821012 548275673 FOR 10000000000 - US Tax",-5,
```
на $42.95 из тех $56.47 было куплено 5 MAC:
```
Trades,Data,Order,Stocks,USD,MAC,"2020-06-03, 09:50:04",5,8.59,8.59,-42.95,0,42.95,0,0,O
```
и в итоге баланс на счете изменился на `$56.47 - $5 - $42.95 = $8.52`?

И на самом деле, если удалить из отчета строчку с корпоративным действием, то он начинает парситься без проблем, и программа не ругается на то, что у нее где-то что-то разъехалось?

> Прислать весь отчет, к сожалению, не могу. Но нужные строчки готов предоставить

Будет здорово, если сможете приложить хотя бы результат `grep MAC statement.csv` - я так хотя бы смогу руками собрать фейковый отчет и сделать регрессионные тесты на него.
